SingularityUI/app/models/Request.coffee

```diff
@@ -466,17 +474,48 @@ class Request extends Model
                             history.push(@data.commandLineInput)
                             localStorage.setItem(@localStorageCommandLineInputKeyPrefix + @id, JSON.stringify(history))
                     localStorage.setItem('taskRunRedirectFilename', fileName) if filename?
-                    localStorage.setItem('taskRunAutoTail', @data.autoTail)
+                    localStorage.setItem('taskRunAfterStart', @data.afterStart)
                     @data.id = @get 'id'
 
-                    @run( @data.commandLineInput, message ).done callback( @data )
+                    if @data.afterStart in ['browse-to-sandbox', 'autoTail']
+                        @data.runId = uuid.v4()
+
+                    doneFn = =>
+                        if @data.afterStart is 'autoTail'
+                            taskPoller = new TaskPoller({
+                                requestId: @id
+                                autoTailFilename: @data.filename
+                                taskPollTimestamp: +new Date()
+                                pollingType: 'autoTail'
+                            })
+
+                            taskPoller.startTaskPolling()
+                        else if @data.afterStart is 'browse-to-sandbox'
+                            taskPoller = new TaskPoller({
+                                requestId: @id
+                                taskPollTimestamp: +new Date()
+                                pollingType: 'browse-to-sandbox'
+                            })
+
+                            taskPoller.startTaskPolling()
+
+                        callback( @data )
+
+                    @run( @data.commandLineInput, message, @data.runId ).done doneFn
                     return true
 
             afterOpen: =>
-                $('#filename').val localStorage.getItem('taskRunRedirectFilename')
-                $('#autoTail').prop 'checked', (localStorage.getItem('taskRunAutoTail') is 'on')
+                taskRunAfterStart = localStorage.getItem('taskRunAfterStart')
+                $('#filename').val localStorage.getItem('taskRunRedirectFilename') or 'service.log'
```

Comment:
Oh, I got that from the placeholder text, which is currently 'e.g. service.log'. But that's a constant, not configurable atm.